src/main/java/com/salesforce/dataloader/controller/Controller.java

```diff
@@ -273,6 +284,66 @@ private void copyFile(File file, File destFile) {
         }
     }
 
+    /**
+     * Create directory provided from the parameter
+     *
+     * @param dirPath - directory to be created
+     *
+     * @return
+     *  True if directory was created successfully or directory already existed
+     *  False if directory was failed to create
+     */
+    private boolean createDir(File dirPath) {
+        boolean isSuccessful = true;
+        if (!dirPath.exists() || !dirPath.isDirectory()) {
+            isSuccessful = dirPath.mkdirs();
+            if (isSuccessful) {
+                logger.info("Created config directory: " + dirPath);
+            } else {
+                logger.info("Unable to create config directory: " + dirPath);
+            }
+        } else {
+            logger.info("Config directory already exists: " + dirPath);
+        }
+        return isSuccessful;
+    }
+
+    /**
+     * Returns current user's Dataloader configuration directory
+     *  ie) For Windows - C:\Users\{user}\AppData\Roaming\salesforce.com\Data Loader {version}\conf
+     *      For Mac - /Users/{user}/Library/Preferences/salesforce.com/Data Loader {version}/conf
+     *      For Unix* - {user.home}/Library/Preferences/salesforce.com/Data Loader {version}/conf
```

Comment:
These are where sensitive files will be stored

Reply:
Yes I agree it's not how Unix/Linux would store user specific files.  Knowing that Unix/linux is not officially supported for DL, I didn't cover the case in detail. I will take care of it.